src/lib/datepicker/calendar.ts

```diff
@@ -228,51 +309,17 @@ export class MatCalendar<D> implements AfterContentInit, OnDestroy, OnChanges {
 
   /** Handles year/month selection in the multi-year/year views. */
   _goToDateInView(date: D, view: 'month' | 'year' | 'multi-year'): void {
-    this._activeDate = date;
-    this._currentView = view;
-  }
-
-  /** Handles user clicks on the period label. */
-  _currentPeriodClicked(): void {
-    this._currentView = this._currentView == 'month' ? 'multi-year' : 'month';
-  }
-
-  /** Handles user clicks on the previous button. */
-  _previousClicked(): void {
-    this._activeDate = this._currentView == 'month' ?
-        this._dateAdapter.addCalendarMonths(this._activeDate, -1) :
-        this._dateAdapter.addCalendarYears(
-            this._activeDate, this._currentView == 'year' ? -1 : -yearsPerPage);
-  }
-
-  /** Handles user clicks on the next button. */
-  _nextClicked(): void {
-    this._activeDate = this._currentView == 'month' ?
-        this._dateAdapter.addCalendarMonths(this._activeDate, 1) :
-        this._dateAdapter.addCalendarYears(
-            this._activeDate, this._currentView == 'year' ? 1 : yearsPerPage);
-  }
-
-  /** Whether the previous period button is enabled. */
-  _previousEnabled(): boolean {
-    if (!this.minDate) {
-      return true;
-    }
-    return !this.minDate || !this._isSameView(this._activeDate, this.minDate);
-  }
-
-  /** Whether the next period button is enabled. */
-  _nextEnabled(): boolean {
-    return !this.maxDate || !this._isSameView(this._activeDate, this.maxDate);
+    this.activeDate = date;
+    this.currentView = view;
   }
 
   /** Whether the two dates represent the same view in the current view mode (month or year). */
-  private _isSameView(date1: D, date2: D): boolean {
-    if (this._currentView == 'month') {
+  isSameView(date1: D, date2: D): boolean {
```

Comment:
is this used in `MatCalendar` anymore? If not you can move it to `MatCalendarHeader` as a private method

Reply:
done in 715e413cd27043b92fb8ba96f741dce1f6e96afd

Reply:
Since now also  a type parameter is used for `MatCalendarHeader`, the signature of `_isSameAs` can stay as it is.